### PR TITLE
fix(environment): add an environment exists api

### LINKED ISSUE #1543: [Bug]: Failed to create custom environment
### ODC version

odc424

### OB version

*

### What happened?

image

### What did you expect to happen?

Created successfully

### How can we reproduce it (as minimally and precisely as possible)?

1. Modify SQL window specifications in the default environment
2. Referencing default environment when creating custom environment

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
1. Adds an API for checking if the environment name exists. 
2. Updating environment failed because I used `@Builder` on `UpdateEnvironmentReq` but not also used `@AllArgsConstructor` and `@NoArgsConstructor`
3. Use `ObjectUitl#copy` instead of returning the original entity for database operation.

#### Which issue(s) this PR fixes:

Fixes #1543 #1544 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yizhouxw — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/environment/EnvironmentService.java [hunk 1]
```diff
@@ -109,11 +109,10 @@ public void init() {
     @Transactional(rollbackFor = Exception.class)
     @PreAuthenticate(actions = "create", resourceType = "ODC_ENVIRONMENT", isForAll = true)
     public Environment create(@NotNull @Valid CreateEnvironmentReq req) {
-        PreConditions.validNoDuplicated(ResourceType.ODC_ENVIRONMENT, "name", req.getName(),
-                () -> exists(req.getName()));
-        PreConditions.validArgumentState(!DEFAULT_ENV_NAMES.contains(req.getName()), ErrorCodes.ReservedName,
-                new Object[] {req.getName()}, "The environment name is not allowed");
-
+        EnvironmentExists exists = exists(req.getName());;
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/environment/EnvironmentService.java:112]: typo: `;;`

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by yhilmare — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/environment/EnvironmentService.java (hunk 1 above)
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/environment/EnvironmentService.java]: 👍🏻

**Review by smallsheeeep — APPROVED** [review]

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by smallsheeeep — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**